### PR TITLE
Update s4 support

### LINKED ISSUE #1223: Enable Intel 2022 and update test support on S4
## Description
Support for S4 should be maintained so users of that system can continue to use the latest features of the UFS.  Currently, S4 support is lagging in two areas: regression testing and Intel 2022 support.

## Solution
1. Update the resources in `default_vars.sh` to reenable regression testing on S4, following #320.
2. Rebuild the hpc-stack libraries on S4 using the Intel 2022.1 compiler suite and update the `ufs_s4.intel` and `ufs_s4.intel_debug` to point to the 2022.1 libraries, following #1113.

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] N/A Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] N/A New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

This updates the core counts and geometries for regression testing on S4.  It also updates the S4 module files to use the Intel 2022.1 compilers and the associated hpc-stack libraries..

### Issue(s) addressed

Fixes #1223

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [ ] CI

## Dependencies

N/A

### CODE REVIEW

**Comment by jkbk2004** [comment]
@DavidHuber-NOAA It will be practical to combine this PR to #1239. Could you make a PR to Dusan's #1239? so that we can merge together.

**Comment by DavidHuber-NOAA** [comment]
@jkbk2004 Yes, I will do that now.

File: tests/default_vars.sh [hunk 1]
```diff
@@ -168,18 +168,16 @@ elif [[ $MACHINE_ID = jet.* ]]; then
...
-  TASKS_stretch=48 ; TPN_stretch=12 ; INPES_stretch=2 ; JNPES_stretch=4
-  TASKS_strnest=96 ; TPN_strnest=12 ; INPES_strnest=2 ; JNPES_strnest=4
+  TASKS_dflt=150 ; INPES_dflt=3 ; JNPES_dflt=8
+  TASKS_thrd=78  ; INPES_thrd=3 ; JNPES_thrd=4
+  TASKS_c384=336 ; INPES_c384=6 ; JNPES_c384=8 ; THRD_c384=2
+  TASKS_c768=928 ; INPES_c768=8 ; JNPES_c768=16 ; THRD_c768=1
 
-  TASKS_cpl_atmw_gdas=560; TPN_cpl_atmw_gdas=16; INPES_cpl_atmw_gdas=6; JNPES_cpl_atmw_gdas=8
-  THRD_cpl_atmw_gdas=2; WPG_cpl_atmw_gdas=24; APB_cpl_atmw_gdas="0 311"; WPB_cpl_atmw_gdas="312 559"
+  TASKS_cpl_atmw_gdas=544; THRD_cpl_atmw_gdas=2
```
Comment (DeniseWorthen) [inline tests/default_vars.sh]: I think this should be 560, not 544: TASKS_cpl_atmw_gdas=560

Comment (DavidHuber-NOAA) [reply tests/default_vars.sh]: Thanks for the catch @DeniseWorthen.  I'll change it and give that test a go.

Comment (DavidHuber-NOAA) [reply tests/default_vars.sh]: Fixed.  The control_c384gdas_wav test passes regression testing.

**Comment by junwang-noaa** [comment]
The PR was merged in #1239. The PR will be closed.

**Pull request closed without merging**